### ISSUE TITLE
calcite-text-area: width not calculated correctly

### ISSUE THREAD

**Issue by AdelheidF** [Esri/calcite-design-system#8654]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Narrow shell area
![image](https://github.com/Esri/calcite-design-system/assets/44481292/8f5933d7-8aff-41e1-ab30-08d82eaf81a3)

Wide shell area
![image](https://github.com/Esri/calcite-design-system/assets/44481292/e6491c8d-849b-4e4a-bcca-d8d5c147b9d1)

### Expected Behavior

input and text-area right border always align vertically

### Reproduction Sample

https://codepen.io/afreitag/pen/WNmXZPm

### Reproduction Steps

Open codepen and look

### Reproduction Version

2.3.0

### Regression?

don't think so

### Priority impact

p2 - want for current milestone

### Impact

won't use text-area until this is fixed

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Map Viewer

**Comment by jcfranco** [comment]
@anveshmekala Can you take a look at the estimate and adjust if needed?

**Comment by anveshmekala** [comment]
> @anveshmekala Can you take a look at the estimate and adjust if needed?

The current estimate looks good.

**Comment by macandcheese** [comment]
You can also drag the text-area to be wider than the width of it's parent. I'd expect it to only extend to the parent's extent. Not sure if this issue is appropriate but mentioning it here.

https://github.com/Esri/calcite-design-system/assets/4733155/0f0299eb-39c1-4310-9334-6bd135f1526a

**Comment by anveshmekala** [comment]
> You can also drag the text-area to be wider than the width of it's parent. I'd expect it to only extend to the parent's extent. Not sure if this issue is the appropriate place but mentioning it here.
> 
Thanks for bring this up. This is consistent with native `<textarea/>` element https://codepen.io/ANVESHMEKALA/pen/rNbmyjz. Workaround is to add max-width property on `<textarea/>` element inside shadowDOM of `calcite-text-area` to restrict the resizing behavior.  We can provide this option via `@prop` may be. Thoughts?

**Comment by macandcheese** [comment]
Yeah, I generally do like to default to matching native functionality - but not sure if it is a benefit to a user here or it could be an improvement our component has there. Could also be a discussion for a separate issue if it's out of scope here.

**Comment by anveshmekala** [comment]
calcite-text-area width is set to `100%` by default which occupies any available space in the container where as calcite-input defaults to min-width. Both calcite-text-area & calcite-input do not align vertically by default _(consistent with native textarea & input [ref](https://codepen.io/anvesh-mekala/pen/jENZJrx))_  when wrapped inside a flex container due to difference in min-width unless specified. 

To align vertically, calcite-text-area width is set to `auto` & `--calcite-text-area-min-width` token to `auto` or` 0` depending on how small can text-area shrink. 

Attached Codepen https://codepen.io/anvesh-mekala/pen/wBvpzWJ

**Comment by DitwanP** [comment]
Closing out as a result of the context and working example that Anvesh provided above 👆.
